code/game/gamemodes/cult/talisman.dm

```diff
@@ -328,9 +328,20 @@
 		if(target.use(25))
 			new /obj/structure/constructshell(T)
 			user << "<span class='warning'>The talisman clings to the metal and twists it into a construct shell!</span>"
+			user << sound('sound/effects/magic.ogg',0,1,25)
+			PoolOrNew(/obj/effect/overlay/temp/cult/turf/open/floor, T)
 			qdel(src)
-		else
-			user << "<span class='warning'>The talisman requires at least 25 sheets of metal!</span>"
+	if(proximity_flag && istype(target, /obj/item/stack/sheet/plasteel) && iscultist(user))
```

Comment:
the proc is already checking proximity and iscultist() above. You can probably do:
```
if(prox and iscultist)
	if(istype(X, metal sheet)) //and change the first argument to just atom/target, we only typecast an argument when there's a single type that's actually useful for the proc, here there's two now (metal and plasteel.
	else if(istype(X, plasteel))
	else
		the talisma requires metal or plasteel
```